mmdet/evaluation/metrics/voc_metric.py

```diff
@@ -91,85 +72,60 @@ def process(self, data_batch: dict, data_samples: Sequence[dict]) -> None:
                 bboxes=gt_instances['bboxes'].cpu().numpy(),
                 bboxes_ignore=gt_ignore_instances['bboxes'].cpu().numpy(),
                 labels_ignore=gt_ignore_instances['labels'].cpu().numpy())
+            groundtruths.append(ann)
 
             pred = data_sample['pred_instances']
-            pred_bboxes = pred['bboxes'].cpu().numpy()
-            pred_scores = pred['scores'].cpu().numpy()
-            pred_labels = pred['labels'].cpu().numpy()
-
-            dets = []
-            for label in range(len(self.dataset_meta['CLASSES'])):
-                index = np.where(pred_labels == label)[0]
-                pred_bbox_scores = np.hstack(
-                    [pred_bboxes[index], pred_scores[index].reshape((-1, 1))])
-                dets.append(pred_bbox_scores)
+            pred['bboxes'] = pred['bboxes'].cpu().numpy()
+            pred['scores'] = pred['scores'].cpu().numpy()
+            pred['labels'] = pred['labels'].cpu().numpy()
+            predictions.append(pred)
 
-            self.results.append((ann, dets))
+        self.add(predictions, groundtruths)
 
-    def compute_metrics(self, results: list) -> dict:
-        """Compute the metrics from processed results.
-
-        Args:
-            results (list): The processed results of each batch.
+    def evaluate(self, *args, **kwargs) -> dict:
+        """Returns metric results and print pretty table of metrics per class.
 
-        Returns:
-            dict: The computed metrics. The keys are the names of the metrics,
-            and the values are corresponding results.
+        This method would be invoked by ``mmengine.Evaluator``.
         """
-        logger: MMLogger = MMLogger.get_current_instance()
-        gts, preds = zip(*results)
-        eval_results = OrderedDict()
-        if self.metric == 'mAP':
-            assert isinstance(self.iou_thrs, list)
-            dataset_type = self.dataset_meta.get('DATASET_TYPE')
-            if dataset_type in ['VOC2007', 'VOC2012']:
```

Comment:
Does this logic still keep in mmeval?

Reply:
`dataset_type` is used to choose [which `tpfp_fn` to use](https://github.com/open-mmlab/mmdetection/blob/e63459234d27b90a770e0a4bcc126b4371e5d2ad/mmdet/evaluation/functional/mean_ap.py#L614) and [get the class names for print map summary](https://github.com/open-mmlab/mmdetection/blob/e63459234d27b90a770e0a4bcc126b4371e5d2ad/mmdet/evaluation/functional/mean_ap.py#L771).

MMEval is no need to choose `tpfp_fn` from inputs now and get the class names from `dataset_meta['CLASSES']` directly. So I remove this stuff.

Reply:
Update a simple check of dataset type and eval mode